_test/test_sqw_class/test_smooth.m

```diff
@@ -0,0 +1,233 @@
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_array_width_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth([100, 25], 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_resolution_shape_arg(obj)
+            skipTest('No valid agruments possible for resolution call #628');
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth([100, 201, 301], 'resolution');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_hat_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth(100, 'hat');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_gaussian_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            d = s.smooth(100, 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_raises_error_with_invalid_shape_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            assertExceptionThrown(@()s.smooth(100, 'not-shape'), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_raises_error_with_incorrect_dimension_width_arg(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s.data.pix = PixelData();
+
+            assertExceptionThrown(@()s.smooth([10,10,10,10,10]), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_raises_error_with_sqw_containing_pixel_data(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+
+            assertExceptionThrown(@()s.smooth(10), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_raises_error_if_any_sqw_has_pixel_data_array_call(obj)
+            s = sqw(obj.test_sqw_2d_fullpath);
+            s_nopix = copy(s);
+            s_nopix.data.pix = PixelData();
+
+            assertExceptionThrown(@() smooth([s_nopix, s, s_nopix], 10), 'HORACE:smooth:invalid_arguments');
+        end
+
+
+        %% SMOOTH_UNITS
+        function test_smooth_units_returns_d2d_object(obj)
```

Comment:
this one is returning a sqw object (and should since it is testing the sqw api) please change the function name

Reply:
oops